src/dev/TPS2HB50BQ1.cpp

```diff
@@ -0,0 +1,143 @@
+    EN2.writePin(IO::GPIO::State::LOW);
+}
+void TPS2HB50BQ1::enable_ps_one() {
+    EN1.writePin(IO::GPIO::State::HIGH);
+}
+
+void TPS2HB50BQ1::disable_ps_one() {
+    EN1.writePin(IO::GPIO::State::LOW);
+}
+
+void TPS2HB50BQ1::enable_ps_two() {
+    EN2.writePin(IO::GPIO::State::HIGH);
+}
+
+void TPS2HB50BQ1::disable_ps_two() {
+    EN2.writePin(IO::GPIO::State::LOW);
+}
+
+void TPS2HB50BQ1::enableDiag() {
+    DIAG_EN.writePin(IO::GPIO::State::HIGH);
+}
+
+void TPS2HB50BQ1::disableDiag() {
+    DIAG_EN.writePin(IO::GPIO::State::LOW);
+}
+
+/**
+ * See TP2HB50BQ1 datasheet for more information on latch modes.
+ * According to datasheet, if latch is low, then switch will auto-retry.
+ * If latch is high, then switch will latch until latch pin is pulled low.
+ * Page 23/27 of the datasheet has a table with the different modes.
+ * @param mode
+ */
+void TPS2HB50BQ1::setLatch(LATCH_MODE mode) {
+    if (mode == LATCH_MODE::LATCHED) {
+        LATCH.writePin(IO::GPIO::State::LOW);
+        current_latch_mode = LATCH_MODE::LATCHED;
+    } else if (mode == LATCH_MODE::AUTO_RETRY) {
+        LATCH.writePin(IO::GPIO::State::HIGH);
+        current_latch_mode = LATCH_MODE::AUTO_RETRY;
+    }
+}
+
+void TPS2HB50BQ1::readSenseOut(uint32_t& senseOut) {
+    senseOut = ADC.readRaw();
+}
+
+/**
+ * Set the diagnostic mode:
+ * DIAG_MODE_OFF: Sets diagnostics pin to low, along with both diag select pins.
+ * DIAG_MODE_FAULT_STATUS: Sets diagnostics pin to high, along with the fault status diag select pin.
+ * DIAG_MODE_CURRENT: Sets diagnostics pin to high, along with the current diag select pin.
+ * DIAG_MODE_TEMP: Sets diagnostics pin to high, along with the temp diag select pin.
+ *
+ * TODO: MUX IS NOT NECESSARILY CORRECT
+ *
+ * @param diag_mode the mode to set the diagnostics to; see above for options, or header file for consts.
+ */
+void TPS2HB50BQ1::setDiagnostics(DIAG_MODE diag_mode) {
+    current_diag_mode = diag_mode;
+    switch (diag_mode) {
+    case OFF:
+        disableDiag();
+        DIAG_SELECT_1.writePin(IO::GPIO::State::LOW);
+        DIAG_SELECT_2.writePin(IO::GPIO::State::LOW);
+        break;
+    case FAULT_STATUS:
+        enableDiag();
+        DIAG_SELECT_1.writePin(IO::GPIO::State::HIGH);
+        DIAG_SELECT_2.writePin(IO::GPIO::State::LOW);
+        break;
+    case CURRENT:
+        enableDiag();
+        DIAG_SELECT_1.writePin(IO::GPIO::State::LOW);
+        DIAG_SELECT_2.writePin(IO::GPIO::State::HIGH);
+        break;
+    case TEMP:
+        enableDiag();
+        DIAG_SELECT_1.writePin(IO::GPIO::State::HIGH);
+        DIAG_SELECT_2.writePin(IO::GPIO::State::HIGH);
+        break;
+    default:
+        break;
+    }
+}
+
+/**
+ * Get the fault status of the TPS2HB50BQ1
+ * Currently do not know the adc output for this, so it is a placeholder.
+ * @return the fault status of the TPS2HB50BQ1
+ */
+uint32_t TPS2HB50BQ1::getCurrent() {
+    if (current_diag_mode != DIAG_MODE::CURRENT) {
```

Comment:
If you always set diagnostics to off at the end of these functions, this if statement will always succeed, so there's not much point to having it here